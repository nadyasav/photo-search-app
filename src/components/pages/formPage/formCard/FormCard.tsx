import React from 'react';
import { IFormData } from '../../../../types/customTypes';
import styles from './FormCard.module.css';

function FormCard(props: { data: IFormData; index: number }) {
  return (
    <div className={styles.card} data-testid={'formCard' + props.index}>
      <ul>
        <li className={styles.card__item}>
          <span className={styles.card__title}>Profile picture:</span>
          <div className={styles.card_img__box}>
            <img src={props.data.file} className={styles.card_img} />
          </div>
        </li>
        <li className={styles.card__item}>
          <span className={styles.card__title}>First Name:</span>
          <span>{props.data.firstName}</span>
        </li>
        <li className={styles.card__item}>
          <span className={styles.card__title}>Last Name:</span>
          <span>{props.data.lastName}</span>
        </li>
        <li className={styles.card__item}>
          <span className={styles.card__title}>Date Of Birth:</span>
          <span>{props.data.dateOfBirth}</span>
        </li>
        <li className={styles.card__item}>
          <span className={styles.card__title}>Country:</span>
          <span>{props.data.country}</span>
        </li>
        <li className={styles.card__item}>
          <span className={styles.card__title}>Gender:</span>
          <span>{props.data.gender}</span>
        </li>
        <li className={styles.card__item}>
          <span className={styles.card__title}>Processing personal data:</span>
          <span>{props.data.agree ? 'Yes' : 'No'}</span>
        </li>
      </ul>
    </div>
  );
}

export default FormCard;
