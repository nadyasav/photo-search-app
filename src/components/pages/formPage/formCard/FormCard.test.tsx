import { render, screen } from '@testing-library/react';
import React from 'react';
import FormCard from './FormCard';

const formData = {
  firstName: '<firstName>',
  lastName: '<lastName>',
  dateOfBirth: '<dateOfBirth>',
  country: '<country>',
  file: '<file>',
  gender: '<gender>',
  agree: '<agree>',
};

describe('Testing FormCard ', () => {
  it('renders FormCard', () => {
    const formCard = render(<FormCard data={formData} index={0} />);
    expect(formCard).toMatchSnapshot();
  });
});
