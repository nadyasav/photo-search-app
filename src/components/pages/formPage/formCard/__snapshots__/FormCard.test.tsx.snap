// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FormCard  renders FormCard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        data-testid="formCard0"
      >
        <ul>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              Profile picture:
            </span>
            <div
              class="card_img__box"
            >
              <img
                class="card_img"
                src="<file>"
              />
            </div>
          </li>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              First Name:
            </span>
            <span>
              &lt;firstName&gt;
            </span>
          </li>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              Last Name:
            </span>
            <span>
              &lt;lastName&gt;
            </span>
          </li>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              Date Of Birth:
            </span>
            <span>
              &lt;dateOfBirth&gt;
            </span>
          </li>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              Country:
            </span>
            <span>
              &lt;country&gt;
            </span>
          </li>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              Gender:
            </span>
            <span>
              &lt;gender&gt;
            </span>
          </li>
          <li
            class="card__item"
          >
            <span
              class="card__title"
            >
              Processing personal data:
            </span>
            <span>
              Yes
            </span>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      data-testid="formCard0"
    >
      <ul>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            Profile picture:
          </span>
          <div
            class="card_img__box"
          >
            <img
              class="card_img"
              src="<file>"
            />
          </div>
        </li>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            First Name:
          </span>
          <span>
            &lt;firstName&gt;
          </span>
        </li>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            Last Name:
          </span>
          <span>
            &lt;lastName&gt;
          </span>
        </li>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            Date Of Birth:
          </span>
          <span>
            &lt;dateOfBirth&gt;
          </span>
        </li>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            Country:
          </span>
          <span>
            &lt;country&gt;
          </span>
        </li>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            Gender:
          </span>
          <span>
            &lt;gender&gt;
          </span>
        </li>
        <li
          class="card__item"
        >
          <span
            class="card__title"
          >
            Processing personal data:
          </span>
          <span>
            Yes
          </span>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
